hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneFSInputStream.java

```diff
@@ -199,4 +203,56 @@ public void readFully(long position, ByteBuffer buf) throws IOException {
       }
     }
   }
+
+  /**
+   * Implements vectored read by reading each range asynchronously.
+   * This allows clients to read multiple byte ranges from the same file
+   * in a single call, potentially improving performance by enabling
+   * parallel reads and reducing round-trip overhead.
+   *
+   * @param ranges list of file ranges to read
+   * @param allocate function to allocate ByteBuffer for each range
+   * @throws IOException if there is an error performing the reads
+   */
+  @Override
+  public void readVectored(List<? extends FileRange> ranges,
+                           IntFunction<ByteBuffer> allocate) throws IOException {
+    TracingUtil.executeInNewSpan("OzoneFSInputStream.readVectored", () -> {
+      // Perform vectored read using positioned read operations
+      for (FileRange range : ranges) {
+        CompletableFuture<ByteBuffer> result = range.getData();
+        if (result == null) {
+          result = new CompletableFuture<>();
+          range.setData(result);
+        }
+
+        final CompletableFuture<ByteBuffer> finalResult = result;
+        final long offset = range.getOffset();
+        final int length = range.getLength();
+
+        // Submit async read task for this range
+        CompletableFuture.runAsync(() -> {
+          try {
+            ByteBuffer buffer = allocate.apply(length);
+            int bytesRead = read(offset, buffer);
+
+            if (bytesRead < length) {
+              finalResult.completeExceptionally(
+                  new EOFException("Requested " + length +
+                      " bytes but only read " + bytesRead));
+            } else {
+              buffer.flip();
+              if (statistics != null) {
+                statistics.incrementBytesRead(bytesRead);
+              }
+              finalResult.complete(buffer);
+            }
+          } catch (Exception e) {
+            finalResult.completeExceptionally(e);
+          }
+        });
+      }
```

Comment:
This almost duplicates the implementation in `MultipartInputStream`.  Can they share code?